xarray/backends/netCDF4_.py

```diff
@@ -414,13 +416,25 @@ def _acquire(self, needs_lock=True):
     def ds(self):
         return self._acquire()
 
-    def open_store_variable(self, name, var):
+    def open_store_variable(self, name: str, var):
+        import netCDF4
+
         dimensions = var.dimensions
-        data = indexing.LazilyIndexedArray(NetCDF4ArrayWrapper(name, self))
         attributes = {k: var.getncattr(k) for k in var.ncattrs()}
+        data = indexing.LazilyIndexedArray(NetCDF4ArrayWrapper(name, self))
+        encoding = {}
+        if isinstance(var.datatype, netCDF4.EnumType):
+            encoding["dtype"] = np.dtype(
+                data.dtype,
+                metadata={
+                    "enum_dict": var.datatype.enum_dict,
```

Comment:
Just a nit, but can we use `enum` instead of `enum_dict`? Then this works for h5py/h5netcdf, too. 

Note: I've explored this setting for h5netcdf and it does need some implementation but should be straightforward. Thanks for pushing this forward.